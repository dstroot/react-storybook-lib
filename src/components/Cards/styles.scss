.card-display {
  font-size: calc(28px + 1.5vw + 0.5vmin);
  font-weight: 300;
}
