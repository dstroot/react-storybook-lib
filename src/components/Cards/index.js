import React from "react";

// styles
import "./styles.scss";

// images
import card1 from "./media/card1.jpg";
import card2 from "./media/card2.jpg";
import card3 from "./media/card3.jpg";
import { ReactComponent as Clock } from "./media/clock.svg";

const Cards = () => (
  <div className="container-fluid bg-light pb-5">
    <div className="row">
      <div className="col text-center p-4">
        <h1 className="card-display">INSIGHTS TO HELP YOU MEET YOUR GOALS</h1>
      </div>
    </div>
    <div className="row">
      <div className="card-deck">
        <div className="card">
          <img className="card-img-top" src={card1} alt="retirement planning" />
          <div className="card-body">
            <h5 className="card-title text-muted">RETIREMENT PLANNING</h5>
            <h3>
              Make the Most of Your Retirement With an Informed Income Strategy.
            </h3>
          </div>
          <div className="card-footer bg-white">
            <a href="/" className="btn btn-primary">
              Read
            </a>
            <div className="d-inline p-2">
              <Clock height="15px" width="15px" />
              <strong className="align-middle"> 5 min read</strong>
            </div>
          </div>
        </div>
        <div className="card">
          <img className="card-img-top" src={card2} alt="business" />
          <div className="card-body">
            <h5 className="card-title text-muted">CAREER AND BUSINESS</h5>
            <h3>Donâ€™t Put Off Saving for Retirement</h3>
            <p className="card-text">
              Start building your nest egg early to prepare for retirement.
            </p>
          </div>
          <div className="card-footer bg-white">
            <a href="/" className="btn btn-primary">
              Read
            </a>
            <div className="d-inline p-2">
              <Clock height="15px" width="15px" />
              <strong className="align-middle"> 7 min read</strong>
            </div>
          </div>
        </div>
        <div className="card">
          <img className="card-img-top" src={card3} alt="older couple" />
          <div className="card-body">
            <h5 className="card-title text-muted">RETIREMENT PLANNING</h5>
            <h3>Deciding When to Claim Retirement Benefits</h3>
            <p className="card-text">
              Claiming benefits when you are younger than your full retirement
              age can mean missing out on hundreds of dollars per month.
            </p>
          </div>
          <div className="card-footer bg-white">
            <a href="/" className="btn btn-primary">
              Read
            </a>
            <div className="d-inline p-2">
              <Clock height="15px" width="15px" />
              <strong className="align-middle"> 3 min read</strong>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default Cards;
