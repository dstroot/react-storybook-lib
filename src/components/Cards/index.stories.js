import React from 'react';
import { storiesOf } from '@storybook/react';
// import { Meta, Story, Props } from '@storybook/addon-docs/blocks';
// import { Meta } from '@storybook/addon-docs/blocks';

// components
import Cards from '../Cards';

// Cards section
storiesOf('Pac Life|Cards', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show Cards', () => <Cards />);
