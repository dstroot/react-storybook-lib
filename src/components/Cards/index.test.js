import React from 'react';
import renderer from 'react-test-renderer';
import Cards from '../Cards';

describe('Cards', () => {
  it('it should render', () => {
    const component = renderer.create(<Cards />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
