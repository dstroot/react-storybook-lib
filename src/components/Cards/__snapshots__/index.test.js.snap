// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards it should render 1`] = `
<div
  className="container-fluid bg-light pb-5"
>
  <div
    className="row"
  >
    <div
      className="col text-center p-4"
    >
      <h1
        className="card-display"
      >
        INSIGHTS TO HELP YOU MEET YOUR GOALS
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="card-deck"
    >
      <div
        className="card"
      >
        <img
          alt="retirement planning"
          className="card-img-top"
          src="card1.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title text-muted"
          >
            RETIREMENT PLANNING
          </h5>
          <h3>
            Make the Most of Your Retirement With an Informed Income Strategy.
          </h3>
        </div>
        <div
          className="card-footer bg-white"
        >
          <a
            className="btn btn-primary"
            href="/"
          >
            Read
          </a>
          <div
            className="d-inline p-2"
          >
            <img
              alt="clock"
              height="15px"
              src="clock.svg"
              width="15px"
            />
            <strong
              className="align-middle"
            >
               5 min read
            </strong>
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <img
          alt="business"
          className="card-img-top"
          src="card2.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title text-muted"
          >
            CAREER AND BUSINESS
          </h5>
          <h3>
            Donâ€™t Put Off Saving for Retirement
          </h3>
          <p
            className="card-text"
          >
            Start building your nest egg early to prepare for retirement.
          </p>
        </div>
        <div
          className="card-footer bg-white"
        >
          <a
            className="btn btn-primary"
            href="/"
          >
            Read
          </a>
          <div
            className="d-inline p-2"
          >
            <img
              alt="clock"
              height="15px"
              src="clock.svg"
              width="15px"
            />
            <strong
              className="align-middle"
            >
               7 min read
            </strong>
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <img
          alt="older couple"
          className="card-img-top"
          src="card3.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title text-muted"
          >
            RETIREMENT PLANNING
          </h5>
          <h3>
            Deciding When to Claim Retirement Benefits
          </h3>
          <p
            className="card-text"
          >
            Claiming benefits when you are younger than your full retirement age can mean missing out on hundreds of dollars per month.
          </p>
        </div>
        <div
          className="card-footer bg-white"
        >
          <a
            className="btn btn-primary"
            href="/"
          >
            Read
          </a>
          <div
            className="d-inline p-2"
          >
            <img
              alt="clock"
              height="15px"
              src="clock.svg"
              width="15px"
            />
            <strong
              className="align-middle"
            >
               3 min read
            </strong>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
